File: code/backend/webflux/v3/common-v3/src/main/java/at/bumzack/common/solr/SolrResponseResponse.java
package at.bumzack.common.solr;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class SolrResponseResponse<DOC_TYPE> {

    @JsonProperty
    private Integer numFound;

    @JsonProperty
    private Integer start;

    private List<DOC_TYPE> docs;


    public SolrResponseResponse() {
    }

    public Integer getNumFound() {
        return numFound;
    }

    public void setNumFound(final Integer numFound) {
        this.numFound = numFound;
    }

    public Integer getStart() {
        return start;
    }

    public void setStart(final Integer start) {
        this.start = start;
    }

    public List<DOC_TYPE> getDocs() {
        return docs;
    }

    public void setDocs(final List<DOC_TYPE> docs) {
        this.docs = docs;
    }

    @Override
    public String toString() {
        return "SolrResponseResponse{" +
                "numFound=" + numFound +
                ", start=" + start +
                ", docs=" + docs +
                '}';
    }
}

File: code/backend/csharp/SearchIndexService/SearchIndexService/Solr/SolrResponseResponse.cs
using Newtonsoft.Json;

namespace SearchIndexService.Solr;

public class SolrResponseResponse<DOC_TYPE>
{
    [JsonProperty(PropertyName = "numFound")]
    public int? NumFound { get; set; }

    [JsonProperty(PropertyName = "start")] 
    public int? Start { get; set; }

    [JsonProperty(PropertyName = "docs")] 
    public List<DOC_TYPE> Docs { get; set; }
}